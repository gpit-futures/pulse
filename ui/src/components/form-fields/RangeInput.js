import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Range } from 'rc-slider';
import 'rc-slider/assets/index.css';
import _ from 'lodash/fp';

export default class RangeInput extends PureComponent {
  static propTypes = {
    label: PropTypes.string.isRequired,
    input: PropTypes.shape({
      value: PropTypes.string,
      onChange: PropTypes.func,
    }).isRequired,
  };

  static defaultProps = {
    marks: {
      0: '0',
      10: '10',
      20: '20',
      30: '30',
      40: '40',
      50: '50',
      60: '60',
      70: '70',
      80: '80',
      90: '90',
      100: '100+',
    },
    input: {
      value: [0, 100],
    },
    defaultValue: [0, 100],
  };

  componentDidMount() {
    this.props.input.onChange(this.props.input.value);
  }

  render() {
    const { label, input, marks, defaultValue } = this.props;

    return (
      <div className="form-group">
        <label htmlFor={input.name} className="control-label">{label}</label>
        <div className="wrap-rzslider-search">
          {!_.isEmpty(input.value) ? <Range
            defaultValue={defaultValue}
            marks={marks}
            value={input.value}
            {...input}
          /> : null}
        </div>
      </div>
    )
  }
}
