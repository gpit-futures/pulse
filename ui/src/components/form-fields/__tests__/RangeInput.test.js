import React from 'react';
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15';

import RangeInput from '../RangeInput';

configure({ adapter: new Adapter() });

const testProps = {
  label: 'Test label',
  input: {
    name: 'test-name',
    value: [0, 100],
    onChange: () => {},
  },
};

describe('Component <RangeInput />', () => {
  it('should renders with all props correctly', () => {
    const component = mount(
      <RangeInput
        label={testProps.label}
        input={testProps.input}
      />);

    // expect(component.find('Range')).toHaveLength(1);
    expect(component.find('label')).toHaveLength(1);
    expect(component.find('label.control-label').text()).toEqual(testProps.label);
    expect(component.find('label.control-label').prop('htmlFor')).toEqual(testProps.input.name);


    expect(component.instance().props.label).toEqual(testProps.label);
    expect(component.instance().props.input).toEqual(testProps.input);

    expect(component).toMatchSnapshot();
  });

  it('should renders with all props correctly shallow testing', () => {
    const component = shallow(
      <RangeInput
        label={testProps.label}
        input={{
          name: 'test-name',
          value: [],
          onChange: () => {},
        }}
      />);

    expect(component).toMatchSnapshot();
  });
});
