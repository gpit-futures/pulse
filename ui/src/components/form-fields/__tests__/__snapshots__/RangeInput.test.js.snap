// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <RangeInput /> should renders with all props correctly 1`] = `
<RangeInput
  defaultValue={
    Array [
      0,
      100,
    ]
  }
  input={
    Object {
      "name": "test-name",
      "onChange": [Function],
      "value": Array [
        0,
        100,
      ],
    }
  }
  label="Test label"
  marks={
    Object {
      "0": "0",
      "10": "10",
      "100": "100+",
      "20": "20",
      "30": "30",
      "40": "40",
      "50": "50",
      "60": "60",
      "70": "70",
      "80": "80",
      "90": "90",
    }
  }
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="test-name"
    >
      Test label
    </label>
    <div
      className="wrap-rzslider-search"
    >
      <ComponentEnhancer(Range)
        activeDotStyle={Object {}}
        allowCross={true}
        className=""
        count={1}
        defaultValue={
          Array [
            0,
            100,
          ]
        }
        disabled={false}
        dotStyle={Object {}}
        dots={false}
        handle={[Function]}
        handleStyle={
          Array [
            Object {},
          ]
        }
        included={true}
        marks={
          Object {
            "0": "0",
            "10": "10",
            "100": "100+",
            "20": "20",
            "30": "30",
            "40": "40",
            "50": "50",
            "60": "60",
            "70": "70",
            "80": "80",
            "90": "90",
          }
        }
        max={100}
        min={0}
        name="test-name"
        onAfterChange={[Function]}
        onBeforeChange={[Function]}
        onChange={[Function]}
        prefixCls="rc-slider"
        pushable={false}
        railStyle={Object {}}
        step={1}
        trackStyle={
          Array [
            Object {},
          ]
        }
        value={
          Array [
            0,
            100,
          ]
        }
        vertical={false}
      >
        <div
          className="rc-slider rc-slider-with-marks"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rc-slider-rail"
            style={Object {}}
          />
          <Track
            className="rc-slider-track rc-slider-track-1"
            included={true}
            key="1"
            length={100}
            offset={0}
            style={Object {}}
            vertical={false}
          >
            <div
              className="rc-slider-track rc-slider-track-1"
              style={
                Object {
                  "left": "0%",
                  "visibility": "visible",
                  "width": "100%",
                }
              }
            />
          </Track>
          <Steps
            activeDotStyle={Object {}}
            dotStyle={Object {}}
            dots={false}
            included={true}
            lowerBound={0}
            marks={
              Object {
                "0": "0",
                "10": "10",
                "100": "100+",
                "20": "20",
                "30": "30",
                "40": "40",
                "50": "50",
                "60": "60",
                "70": "70",
                "80": "80",
                "90": "90",
              }
            }
            max={100}
            min={0}
            prefixCls="rc-slider"
            step={1}
            upperBound={100}
            vertical={false}
          >
            <div
              className="rc-slider-step"
            >
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="0"
                style={
                  Object {
                    "left": "0%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="10"
                style={
                  Object {
                    "left": "10%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="20"
                style={
                  Object {
                    "left": "20%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="30"
                style={
                  Object {
                    "left": "30%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="40"
                style={
                  Object {
                    "left": "40%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="50"
                style={
                  Object {
                    "left": "50%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="60"
                style={
                  Object {
                    "left": "60%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="70"
                style={
                  Object {
                    "left": "70%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="80"
                style={
                  Object {
                    "left": "80%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="90"
                style={
                  Object {
                    "left": "90%",
                  }
                }
              />
              <span
                className="rc-slider-dot rc-slider-dot-active"
                key="100"
                style={
                  Object {
                    "left": "100%",
                  }
                }
              />
            </div>
          </Steps>
          <Handle
            className="rc-slider-handle rc-slider-handle-1"
            disabled={false}
            key="0"
            max={100}
            min={0}
            offset={0}
            style={Object {}}
            value={0}
            vertical={false}
          >
            <div
              aria-disabled={false}
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={0}
              className="rc-slider-handle rc-slider-handle-1"
              role="slider"
              style={
                Object {
                  "left": "0%",
                }
              }
              tabIndex="0"
            />
          </Handle>
          <Handle
            className="rc-slider-handle rc-slider-handle-2"
            disabled={false}
            key="1"
            max={100}
            min={0}
            offset={100}
            value={100}
            vertical={false}
          >
            <div
              aria-disabled={false}
              aria-valuemax={100}
              aria-valuemin={0}
              aria-valuenow={100}
              className="rc-slider-handle rc-slider-handle-2"
              role="slider"
              style={
                Object {
                  "left": "100%",
                }
              }
              tabIndex="0"
            />
          </Handle>
          <Marks
            className="rc-slider-mark"
            included={true}
            lowerBound={0}
            marks={
              Object {
                "0": "0",
                "10": "10",
                "100": "100+",
                "20": "20",
                "30": "30",
                "40": "40",
                "50": "50",
                "60": "60",
                "70": "70",
                "80": "80",
                "90": "90",
              }
            }
            max={100}
            min={0}
            upperBound={100}
            vertical={false}
          >
            <div
              className="rc-slider-mark"
            >
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="0"
                style={
                  Object {
                    "left": "0%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                0
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="10"
                style={
                  Object {
                    "left": "10%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                10
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="20"
                style={
                  Object {
                    "left": "20%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                20
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="30"
                style={
                  Object {
                    "left": "30%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                30
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="40"
                style={
                  Object {
                    "left": "40%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                40
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="50"
                style={
                  Object {
                    "left": "50%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                50
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="60"
                style={
                  Object {
                    "left": "60%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                60
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="70"
                style={
                  Object {
                    "left": "70%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                70
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="80"
                style={
                  Object {
                    "left": "80%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                80
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="90"
                style={
                  Object {
                    "left": "90%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                90
              </span>
              <span
                className="rc-slider-mark-text rc-slider-mark-text-active"
                key="100"
                style={
                  Object {
                    "left": "100%",
                    "marginLeft": "-4.5%",
                    "width": "9%",
                  }
                }
              >
                100+
              </span>
            </div>
          </Marks>
        </div>
      </ComponentEnhancer(Range)>
    </div>
  </div>
</RangeInput>
`;

exports[`Component <RangeInput /> should renders with all props correctly shallow testing 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="test-name"
  >
    Test label
  </label>
  <div
    className="wrap-rzslider-search"
  />
</div>
`;
